app/controllers/InstanceRegistryController.scala

```diff
@@ -68,7 +76,195 @@ class InstanceRegistryController @Inject()(myExecutionContext: MyExecutionContex
       } else {
         new Status(response.status)
       }
-    }(myExecutionContext)
+  }(myExecutionContext)
+  }
+
+  //This function is for POST method to the Scala web server
+ 
+
+  def postInstance () : Action[AnyContent] = Action.async { request =>
+  var compType = ""
+  var name = ""
+  
+  request.body.asJson.map { json =>
```

Comment:
please have a look at the usage of the map function, this is generally not the way how to use it.

Reply:
not resolved

Reply:
I have had discussion with the backend team about this. Since, Angular is sending the POST request as a Json array and not as a string the only possible solution I found was to use map. So kept it unchanged in the latest commit.

Reply:
using map is fine I was talking about the way u use the map function here. U basically use it to fill an outside variable. So u use map as a for loop, which is not the intended way.
map is a function which maps the given input to whatever you define in the body of the map function. So what you want to do is assign the result of the map function to an object. describe inside the map function how to map whatever you receive to fill the content of the object and then work with the object in the further flow of the program.
Maybe an example to clear things up:
```
val futureResult: Future[JsResult[Person]] = ws.url(url).get().map {
  response => (response.json \ "person").validate[Person]
}
```

so you have some response object given by a request (which is a json object, same as your body in the post request).
inside the map function the given response is mapped and validated to a person. then the map function returns the content of the map function and assignes it the future result variable.

Reply:
We tried the way you provided the example above but it did not work. When applied the above method it compiles successfully but while executing it throws 'Bad Element' error.

Reply:
I just realized you are reading only the query parameter from the body of the request. you actually don't need to do this at all. 
You don't store any specific information in the post body but send all parameters as query parameters so you should be able to directly access them as I have done in the get requests above. your actual request looks something like /startInstance?InstanceID=5 anyways so you can acess the params directly without parsing the request body.

Reply:
We have tried the way it is done for 'Get ' method long back, but it dint work like that. It used to throw an error (not sure 403 or 404). So we finally had to change it the way we have written now.